packages/opentelemetry-sampler-aws-xray/src/rule-cache.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { SamplingRule } from './sampling-rule';
+import { Attributes } from '@opentelemetry/api';
+import { Resource } from '@opentelemetry/resources';
+
+const RuleCacheTTL_SECONDS = 60 * 60; // The cache expires 1 hour after the last refresh time. (in seconds)
+
+export class RuleCache {
+  public rules: SamplingRule[];
+  private lastUpdated: number | undefined = undefined;
+
+  constructor() {
+    this.rules = [];
+  }
+
+  public isExpired = (): boolean => {
+    if (this.lastUpdated === undefined) {
+      return true;
+    }
+
+    const now = Math.floor(new Date().getTime() / 1000); // getTime returns milliseconds -> /1000 gives seconds
+    return now > this.lastUpdated + RuleCacheTTL_SECONDS;
+  };
+
+  public getMatchedRule = (
+    attributes: Attributes,
+    resource: Resource
+  ): SamplingRule | undefined => {
+    if (this.isExpired()) {
```

Comment:
Do we need to do this check again? We are already checking that cache expired in the remote sampler.shouldSample method.